services_app/src/androidTest/java/org/opendatakit/database/service/OdkDatabaseTypesTest.java

```diff
@@ -136,34 +137,34 @@ private ContentValues contentValuesTestSeti(int i) {
    }
 
    private void verifyRowTestSet1(TypedRow row) {
-      assertEquals(row.getDataByKey(COL_INTEGER_ID), Long.valueOf(TEST_INT_1));
+      assertEquals(row.getDataByKey(COL_INTEGER_ID), (long) TEST_INT_1);
       assertNull(row.getDataByKey(COL_NUMBER_ID));
-      assertEquals(row.getDataByKey(COL_BOOL_ID), Boolean.valueOf(TEST_BOOL_1));
+      assertEquals(row.getDataByKey(COL_BOOL_ID), Boolean.TRUE);
       assertNull(row.getDataByKey(COL_ROWPATH_ID));
       assertNull(row.getDataByKey(COL_CONFIGPATH_ID));
       assertNull(row.getDataByKey(COL_ARRAY_ID));
       assertEquals(row.getDataByKey(COL_STRING_ID), TEST_STR_1);
 
-      assertEquals(row.getDataByKey(COL_GEO_OBJ_ID_ACC), Double.valueOf(TEST_NUM_1));
-      assertEquals(row.getDataByKey(COL_GEO_OBJ_ID_ALT), Double.valueOf(TEST_NUM_1));
-      assertEquals(row.getDataByKey(COL_GEO_OBJ_ID_LAT), Double.valueOf(TEST_NUM_1));
-      assertEquals(row.getDataByKey(COL_GEO_OBJ_ID_LONG), Double.valueOf(TEST_NUM_1));
+      assertEquals(row.getDataByKey(COL_GEO_OBJ_ID_ACC), TEST_NUM_1);
+      assertEquals(row.getDataByKey(COL_GEO_OBJ_ID_ALT), TEST_NUM_1);
+      assertEquals(row.getDataByKey(COL_GEO_OBJ_ID_LAT), TEST_NUM_1);
+      assertEquals(row.getDataByKey(COL_GEO_OBJ_ID_LONG), TEST_NUM_1);
 
    }
 
    private void verifyRowTestSeti(TypedRow row, int i) {
-      assertEquals(row.getDataByKey(COL_INTEGER_ID), Long.valueOf(TEST_INT_i + i));
-      assertEquals(row.getDataByKey(COL_NUMBER_ID), Double.valueOf(TEST_NUM_i + i));
-      assertEquals(row.getDataByKey(COL_BOOL_ID), Boolean.valueOf((i % 2 != 0)));
+      assertEquals(row.getDataByKey(COL_INTEGER_ID), (long) (TEST_INT_i + i));
```

Comment:
row.getDataByKey returns an object. You are removing the conversion to an object so that more casting has to be done. "Long" produces a long oject instead of long.